src/kernels/raw/finder/jupyterPaths.node.ts

```diff
@@ -281,12 +283,27 @@ export class JupyterPaths {
             }
         }
     }
+    private cachedKernelSpecRootPaths?: { promise: Promise<Uri[]>; stopWatch: StopWatch };
     /**
      * This list comes from the docs here:
      * https://jupyter-client.readthedocs.io/en/stable/kernels.html#kernel-specs
      */
-    @traceDecoratorVerbose('Get KernelSpec root path')
     public async getKernelSpecRootPaths(cancelToken: CancellationToken): Promise<Uri[]> {
```

Comment:
Caching the results and logging only when we call without cachine.
Else this results in excessive logging, also with the caching we've improved perf slightly